<h1> Test Exercise 2

In [1]:
data <- read.csv("./data/TestExer2-GPA-round2.csv")
names(data)[1] <- "Observation"

In [3]:
summary(data)

  Observation       FGPA            SATM            SATV            FEM        
 Min.   :  1   Min.   :1.500   Min.   :4.000   Min.   :3.100   Min.   :0.0000  
 1st Qu.:153   1st Qu.:2.485   1st Qu.:5.900   1st Qu.:5.100   1st Qu.:0.0000  
 Median :305   Median :2.773   Median :6.300   Median :5.500   Median :0.0000  
 Mean   :305   Mean   :2.793   Mean   :6.248   Mean   :5.565   Mean   :0.3875  
 3rd Qu.:457   3rd Qu.:3.116   3rd Qu.:6.600   3rd Qu.:6.000   3rd Qu.:1.0000  
 Max.   :609   Max.   :3.971   Max.   :7.900   Max.   :7.600   Max.   :1.0000  

<h3>(a) (i) Regress FGPA on a constant and SATV. Report the coefficient of SATV and its standard error and p-value
(give your answers with 3 decimals).

In [6]:
fit <- lm(FGPA ~ SATV, data=data)
sprintf("Regression line: FGPA = %.2f + %.2fSATV + e", coef(fit)[1], coef(fit)[2])
prinround(t(summary(fit)$coefficie, 3))t)

[1] "Regression line: FGPA = 2.44 + 0.06SATV + e"

            Estimate Std. Error t value Pr(>|t|)
(Intercept)    2.442      0.155  15.747    0.000
SATV           0.063      0.028   2.280    0.023


<h3> (a) (ii) Determine a 95% confidence interval (with 3 decimals) for the effect on FGPA of an increase by 1 point
in SATV.

In [9]:
se = coef(summary(fit))[2, "Std. Error"]
sprintf("Confidence interval at 95%%: (%3f5, %3f5"),  oef(fit)[2]-2cs,e coef(fit)[2]+2*se)

[1] "Confidence interval at 95%: (0.008, 0.118)"

<h3>(b) Answer questions (a-i) and (a-ii) also for the regression of FGPA on a constant, SATV, SATM, and FEM.</h3>

In [19]:
multi_fit <- lm(FGPA ~ SATV+SATM+FEM, data=data)
sprintf("Regression line: FGPA = %.2f + %.2fSATV + %.2fSATM + %.2fFEM + e", coef(multi_fit)[1], coef(multi_fit)[2], coef(multi_fit)[3], coef(multi_fit)[4])  
prit(summary(multi_fit)$coefficien)s)

[1] "Regression line: FGPA = 1.56 + 0.01SATV + 0.17SATM + 0.20FEM + e"

             Estimate Std. Error   t value     Pr(>|t|)
(Intercept) 1.5570482 0.21609551 7.2053704 1.729863e-12
SATV        0.0141619 0.02792697 0.5071047 6.122662e-01
SATM        0.1727359 0.03192671 5.4103874 9.071480e-08
FEM         0.2002716 0.03738085 5.3575989 1.200266e-07


In [21]:
se_SATV = coef(summary(multi_fit))[2, "Std. Error"]
sprintf("Confidence intervalafor SATV coefficienta t 95%%: (%35f, %35f)",  coeffmultif_it)[2]-2*s,_SAT,V coeffmultif_it)[2]+2*s)_SAT)V
se_SATM = coef(summary(multi_fit))[3, "Std. Error"]
sprintf("Confidence interval for SATM coefficient at 95%%: (%.3f, %.3f)",  coef(multi_fit)[3]-2*se_SATM, coef(multi_fit)[3]+2*se_SATM)3
se_FEM = coef(summary(multi_fit))[4, "Std. Error"]
sprintf("Confidence interval for FEM coefficient at 95%%: (%.3f, %.3f)",  coef(multi_fit)[4]-2*se_FEM, coef(multi_fit)[4]+2*se_FEM)se)_FEM

[1] "Confidence interval for SATV coefficient at 95%: (-0.042, 0.070)"

[1] "Confidence interval for SATM coefficient at 95%: (0.109, 0.237)"

[1] "Confidence interval for FEM coefficient at 95%: (0.126, 0.275)"

<h3> (c)  Determine the (4 × 4) correlation matrix of FGPA, SATV, SATM, and FEM. Use these correlations to explain the differences between the outcomes in parts (a) and (b). </h3>

In part (a): $\beta_{SATV}=$ 0.063<br>
In part (b): $\beta_{SATV}=$ 0.014<br><br>

The value of $\beta_{SATV}=$ is affected by its relationship with other factors. We can look at the formula that governs the effect of towards the dependent variable.<br><br>
$\text{Total Effect} = \text{Partial Effect} + \text{Indirect Effect}$<br>
$\frac{dy}{dx_j} = \beta_j+ \sum_{h=2, h\ne j}^k \beta_h\frac{\partial x_h}{\partial x_j}$

Where $\frac{\partial x_h}{\partial x_j}$ is proportional to the correlation between factor of interest (SATV in this case) with other factors.<br>
The above equation means that as we add other factors, the effect of existing factors are explained by their relationship with other factors. Since the total effect for each factor remains the same, $\beta$ of the factor of interest will change.<br><br><br>
From the correlation matrix: <br>
Correlation between SATV and SATM: 0.2878 > 0, $\beta_{SATM} =$ 0.173 > 0<br>
Correlation between SATV and FEM: 0.0336 > 0, $\beta_{FEM} =$ 0.200 > 0<br>
Since total effect remains the same, $\beta_{SATV}$ has a lower value.


In [22]:
print(round(cor(data[, -1]), 4))

       FGPA    SATM   SATV     FEM
FGPA 1.0000  0.1950 0.0922  0.1765
SATM 0.1950  1.0000 0.2878 -0.1627
SATV 0.0922  0.2878 1.0000  0.0336
FEM  0.1765 -0.1627 0.0336  1.0000


<h3>(d) (i) Perform an F-test on the significance (at the 5% level) of the effect of SATV on FGPA, based on the regression in part (b) and another regression.</h3>
Note: Use the F-test in terms of SSR or R2 and use 6 decimals in your computations. The relevant
critical value is 3.9.

$F = \frac{(R_1^2 - R_0^2)/g}{(1-R^2_1)/(n-k)}$<br>
$H_0: \beta_{SATV} = 0, \; H_1: \beta_{SATV} \ne 0$


In [23]:
multi_fit2<- lm(FGPA ~ SATM+FEM, data=data)

In [28]:
sprintf("Regression 0 (another regression) FGPA = %.2f + %.2fSATM + %.2fFEM + e", coef(multi_fit2)[1], coef(multi_fit2)[2], coef(multi_fit2)[3])
srsprintf("Regression 1 (from part (b)): FGPA = %.2f + %.2fSATV + %.2fSATM + %.2fFEM + e", coef(multi_fit)[1], coef(multi_fit)[2], coef(multi_fit)[3], coef(multi_fit)[4]

r) sq_0 = as.numeric(summary(multi_fit)$r2.square)

rsq_1 = as.numeric(summary(multi_fir2t2)$.squaed)

 = 1F n = nrow(ata)

F = = = 40 -  ()rs)q_ - )0*)n()1-s)qr
rsq_1/gf(rnt()Ff""F-statistics va6u",: )%0.5f6 F
"rintf()""F-statisticsue =  < 3-0.2%0.6f.9, therefore do noH_0"jec)t $H_$, F

character(0)

character(0)

<h3> (d) (ii) Check numerically that F = t^2 </h3>

$t = \frac{b_j}{s\sqrt{a_{jj}}}$

In [29]:
b_j <- coef(multi_fit)[2]
se <- coef(summary(multi_fit))[2, 'Std. Error'2)*a_jj)
tsq = (b_j/se)**2

sprintf("F = %0.6f", F)
sprintf("t-squared = %0.6f", tsq)

character(0)

[1] "t-squared = 0.257155"